src/route/seg6.rs

```diff
@@ -0,0 +1,190 @@
+        let segments = match self.mode {
+            Seg6Mode::Inline => self.segments.len() + 1,
+            Seg6Mode::Encap => self.segments.len(),
+            Seg6Mode::Other(_) => self.segments.len(),
+        };
+        12 + 16 * segments
+    }
+
+    fn kind(&self) -> u16 {
+        SEG6_IPTUNNEL_SRH
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        // Some sources for understanding the format of Seg6 in Netlink
+        // https://github.com/torvalds/linux/blob/master/include/uapi/linux/seg6.h
+        // https://github.com/iproute2/iproute2/blob/main/include/uapi/linux/seg6_iptunnel.h#L27
+        // https://github.com/iproute2/iproute2/blob/e3f9681d4a777fb2595a322b421abf0036ab1aae/ip/iproute_lwtunnel.c#L952
+
+        let mut number_segments = self.segments.len();
+        if matches!(self.mode, Seg6Mode::Inline) {
+            number_segments += 1 // last segment (::) added
+        }
+
+        let srhlen = 8 + 16 * number_segments;
+
+        // mode : 4 bytes
+        NativeEndian::write_u32(&mut buffer[..4], self.mode.into());
+        // nexthdr : 1 bytes
+        buffer[4] = 0;
+        // hdrlen : 1 bytes
+        buffer[5] = ((srhlen >> 3) - 1) as u8;
+        // type : 1 byte
+        buffer[6] = 4;
+        // segments_left : 1 byte
+        buffer[7] = (number_segments - 1) as u8;
+        // first_segment : 1 byte
+        buffer[8] = (number_segments - 1) as u8;
+        // flags : 1 byte
+        buffer[9] = 0;
+        // tag : 2 bytes
+        NativeEndian::write_u16(&mut buffer[10..12], 0);
+
+        let mut offset = 12;
+
+        // Add the last segment (::) if working in inline mode
+        if matches!(self.mode, Seg6Mode::Inline) {
+            let addr: Ipv6Addr = "::".parse().expect("Impossible error");
+            buffer[offset..offset + 16].copy_from_slice(&addr.octets());
+            offset += 16;
+        }
+
+        // Add all segments in reverse order
+        for addr in self.segments.iter().rev() {
+            buffer[offset..offset + 16].copy_from_slice(&addr.octets());
+            offset += 16;
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>>
+    for RouteSeg6IpTunnel
+{
+    fn parse(
+        buf: &NlaBuffer<&'a T>,
+    ) -> Result<Self, netlink_packet_utils::DecodeError> {
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            SEG6_IPTUNNEL_SRH => {
+                let mode = NativeEndian::read_u32(payload).into();
+
+                let number_segments = payload[7];
+
+                let mut offset = 12;
+                let mut segments: Vec<Ipv6Addr> = vec![];
+                for _ in 0..number_segments + 1 {
+                    let slice: [u8; 16] = payload[offset..offset + 16]
+                        .try_into()
+                        .expect("Impossible to fail");
+                    let ip_addr = Ipv6Addr::from(slice);
+                    segments.push(ip_addr);
```

Comment:
Same here. Hard coded index is hard to debug and prone to human error. Please use `buffer!()` like `LinkMessageBuffer`.